### PR TITLE
Submit this collection

### PR DESCRIPTION
**Pull request by benbosman**

### CODE REVIEW

File: src/app/+collection-page/collection-page.component.html [hunk 1]
```diff
@@ -32,6 +32,13 @@
                 </ds-comcol-page-content>
 
               </header>
+
+              <div class="add">
+                <a class="btn btn-lg btn-primary mt-1 ml-2" [routerLink]="['/submit']" [queryParams]="{collection:collection.id}" role="button">
```
Comment (artlowel) [inline src/app/+collection-page/collection-page.component.html]: the `mt-1` and `ml-2` add the margin-top and margin-left that cause the button to be weirdly offset compared to the rest of the page. (copy pasted from a different place if I had to guess 😉). 

I'd remove them both and add a `mb-3` (for margin-bottom) instead to put some space between the button and the browse header below.

File: src/app/core/submission/submission-rest.service.ts [hunk 2]
```diff
@@ -65,9 +66,15 @@ export class SubmissionRestService {
...
    *    The identifier for the object
+   * @param collectionId
+   *    The owning collection for the object
    */
-  protected getEndpointByIDHref(endpoint, resourceID): string {
-    return isNotEmpty(resourceID) ? `${endpoint}/${resourceID}` : `${endpoint}`;
+  protected getEndpointByIDHref(endpoint, resourceID, collectionId?: string): string {
+    let url = isNotEmpty(resourceID) ? `${endpoint}/${resourceID}` : `${endpoint}`;
+    if (collectionId) {
+      url = new URLCombiner(url, '?owningCollection=' + collectionId).toString();
```
Comment (artlowel) [inline src/app/core/submission/submission-rest.service.ts]: Not really an issue, just an FYI: in typescript (and post-2015 javascript) [string templating ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)is built in. You can write the same thing like this 
```suggestion
      url = new URLCombiner(url, `?owningCollection=${collectionId}`).toString();
```

File: src/app/+collection-page/collection-page.component.html (hunk 1 above)
Comment (benbosman) [reply src/app/+collection-page/collection-page.component.html]: Thanks, I didn't know what the various classes are, and copied this from the mydspace page
This was a temporary button until I moved it to the navigations, which I've adjusted now

**Comment by benbosman** [comment]
Can we verify what's deployed on https://dspace7.4science.cloud/
https://github.com/DSpace/DSpace/pull/2614/files#diff-ecad0d4d41dd40561d0acfaff2387d57 is required to ensure this works, merged on Dec 16th

I'll verify whether I know how to add a test for the button in the navbar, but not sure whether I'll figure it out because it's a more complicated button

**Comment by artlowel** [comment]
>Can we verify what's deployed on https://dspace7.4science.cloud/
https://github.com/DSpace/DSpace/pull/2614/files#diff-ecad0d4d41dd40561d0acfaff2387d57 is required to ensure this works, merged on Dec 16th

Yes we can. It's based on the branch [rest-demo](https://github.com/DSpace/DSpace/tree/rest-demo). The server automatically updates when changes are pushed to it. It was quite outdated, so I just created and merged [a PR to update it](https://github.com/DSpace/DSpace/pull/2639)

**Pull request merged**